runners/core-java/src/main/java/org/apache/beam/runners/core/StatefulDoFnRunner.java

```diff
@@ -208,7 +212,11 @@ public void setForWindow(InputT input, BoundedWindow window) {
       // make sure this fires after any window.maxTimestamp() timers
       gcTime = gcTime.plus(GC_DELAY_MS);
       timerInternals.setTimer(
-          StateNamespaces.window(windowCoder, window), GC_TIMER_ID, gcTime, TimeDomain.EVENT_TIME);
+          StateNamespaces.window(windowCoder, window),
+          GC_TIMER_ID,
+          gcTime,
+          gcTime,
```

Comment:
The GC timer should have an output timestamp that is the `window.maxTimestamp()`.

Reply:
done